packages/jest-runner/src/run_test.js

```diff
@@ -98,6 +93,13 @@ async function runTestInternal(
     testConsole = new BufferedConsole();
   }
 
+  const environment = new TestEnvironment(
+    Object.assign({}, config, {console: testConsole}),
+  );
+  const leakDetector = config.detectLeaks
+    ? new LeakDetector(environment)
+    : null;
+
   const cacheFS = {[path]: testSource};
   setGlobal(environment.global, 'console', testConsole);
```

Comment:
Are you saying that this function doesn't work anymore in the latest version of jsdom? Can we just fix that instead (using Object.defineProperty?).

Reply:
I'd prefer using the offered API rather than trying to patch it. The way it's done here is a bit ad hoc though. Saying that all custom envs must attach console themselves is pretty breaking, so not sure about best way forward

Reply:
The problem is that `VirtualConsole` gets called with `sendTo` and is passed the `console` object and holds a reference to that (calling it `anyConsole`) rather than referencing `global.console`.

Reply:
I don't understand how either of your comments relate to my question about the `setGlobal` call.

Reply:
Mine is: "JSDOM has an API for setting the console, we should use it rather than just assigning to the global".

Why it stopped working I don't know, though

Reply:
@SimenB but then it would only work for jsdom and every other environment will have to do something different.

Reply:
I'll provide a link when I get to my computer. But I hoped my comment above would explain why the current solution wouldn't ever work...

Reply:
So, it's not a regression. It just never appeared before now. It happened now because of the changes in React. Look closer at my original issue and I explain.

Reply:
When JSDOM is initialized, if we don't provide our own implementation of VirtualConsole, then it uses one which references the global `console`:

https://github.com/tmpvar/jsdom/blob/8a6894c34f14b9131d0fe4acadb71dae4f49daca/lib/api.js#L291-L293

This is all happening outside the test environment, so using `setGlobal` wouldn't work anyway. On top of that, the `sendTo` method called accepts an argument called `anyConsole`. It uses `anyConsole.error` instead of `console.error`. That's here in the VirtualConsole code:

https://github.com/tmpvar/jsdom/blob/5bc6b3bbcc0f18eccecca551bfcfe7d6bbe3e2ba/lib/jsdom/virtual-console.js#L14-L33

This is why we need to initialize JSDOM with our own instance of `VirtualConsole`. One which uses the `testConsole` rather than the global `console`.

I hope that makes sense.

Reply:
Anyway, I'm fine adding this as an additional parameter to the TestEnvironment constructor, but because of how JSDOM constructs a VirtualConsole with a reference to `console` rather than using the global console, we do have to provide our own that references the right console.

What would you prefer?